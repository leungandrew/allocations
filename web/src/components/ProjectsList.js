import React from 'react';
import gql from 'graphql-tag';
import { Query }  from 'react-apollo';

export const GET_PROJECTS_QUERY = gql`
  query getProjects {
    projects {
      name
    }
  }`;

const ProjectsList = () => {
  <Query query={GET_PROJECTS_QUERY}>
    {({loading, error, data}) => {
      if (loading || error) return null;
      return (
        <div>
          <li>Project A</li>
          <li>Project B</li>
        </div>
      )
    }}
  </Query>
};

export default ProjectsList;
