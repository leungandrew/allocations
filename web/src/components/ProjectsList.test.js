import React from 'react';
import ProjectsList, { GET_PROJECTS_QUERY } from './ProjectsList';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';

const mocks = [
  {
    request: {
      query: GET_PROJECTS_QUERY,
    },
    result: {
      data: {
        projects: [
          {
            id: 1,
            name: "Project A",
          },
          {
            id: 2,
             name: "Project B",
          }
        ]
      }
    }
  }
];

describe('ProjectsList', () => {
  it('renders a list of projects', () => {
    const subject = shallow(
      <MockedProvider mocks={mocks}>
        <ProjectsList/>
      </MockedProvider>
        );
    expect(toJson(subject)).toMatchSnapshot();
  });
});
