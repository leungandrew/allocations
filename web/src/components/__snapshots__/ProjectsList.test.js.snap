// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsList renders a list of projects 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "cache": InMemoryCache {
        "addTypename": true,
        "config": Object {
          "addTypename": true,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "storeFactory": [Function],
        },
        "data": ObjectCache {
          "data": Object {},
        },
        "optimistic": Array [],
        "silenceBroadcast": false,
        "typenameDocumentCache": WeakMap {},
        "watches": Array [],
      },
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": ApolloLink {
        "request": [Function],
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "ssrMode": false,
      "store": DataStore {
        "cache": InMemoryCache {
          "addTypename": true,
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "storeFactory": [Function],
          },
          "data": ObjectCache {
            "data": Object {},
          },
          "optimistic": Array [],
          "silenceBroadcast": false,
          "typenameDocumentCache": WeakMap {},
          "watches": Array [],
        },
      },
      "version": "2.3.7",
      "watchQuery": [Function],
    }
  }
>
  <ProjectsList />
</ApolloProvider>
`;
